omod/src/main/java/org/openmrs/module/coreapps/helper/CurrentActiveVisit.java

```diff
@@ -0,0 +1,46 @@
+package org.openmrs.module.coreapps.helper;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.Patient;
+import org.openmrs.Visit;
+import org.openmrs.api.VisitService;
+import org.openmrs.api.context.Context;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * This class is used to get current(latest) active visit for a specific patient.
+ */
+public class CurrentActiveVisit {
```

Comment:
I earlier saw that logic. But because of certain issue I had with it, I decided to go with this one. (For instance, If a patient is currently checked in to another location, other than the current one, this logic may not suffice).
I am saying this because if locations are configured separately on openmrs, and all not part of/under Amani Hospital for instance, this issue may crop up.

But then, I can still replace it with the above logic you suggested. What do you think?